src/Stache/Stores/Store.php

```diff
@@ -27,6 +27,57 @@ abstract class Store
     protected $shouldCacheFileItems = false;
     protected $modified;
     protected $keys;
+    protected $identifiedBy = 'id';
+
+    protected function resolveFromIndex($keys, $column)
+    {
+        return $this->resolveIndex($column)
+            ->load()
+            ->items()
+            ->where(function ($value, $key) use (&$keys) {
+                return $keys->has($key);
+            });
```

Comment:
I've added two tests that will fail (one for the AggregateStore case, and one for the base Store case) here: https://github.com/statamic/cms/pull/9686/commits/b5687a596ed188e1b8dc328b47276c2925428f33 (had wrong files selected so commit message is dumb).

These key checks help to prevent too much data from being returned when the filtered keys contains less items than the index